src/storage/mutate/UpdateEdgeProcessor.cpp

```diff
@@ -171,17 +172,46 @@ kvstore::ResultCode UpdateEdgeProcessor::collectEdgesProps(
         if (constSchema == nullptr) {
             return kvstore::ResultCode::ERR_UNKNOWN;
         }
+        auto schema = std::const_pointer_cast<meta::SchemaProviderIf>(constSchema);
+        auto updater = std::make_unique<RowUpdater>(schema);
+
         for (auto index = 0UL; index < constSchema->getNumFields(); index++) {
             auto propName = std::string(constSchema->getFieldName(index));
-            OptVariantType value = RowReader::getDefaultProp(constSchema.get(), propName);
+            auto findIter = std::find_if(updateItems_.cbegin(), updateItems_.cend(),
+                    [&propName](auto &item) { return item.prop == propName; });
+            OptVariantType value;
+            if (findIter == updateItems_.end()) {
+                value = std::dynamic_pointer_cast<const meta::NebulaSchemaProvider>(
```

Comment:
You could perform the dynamic casting at the first place, instead of casting every time, which is not a fast operation.